### PR TITLE
Offboard Comms Issues

### LINKED ISSUE #40: Fix offboard comms issues
Address all issues here: https://github.com/DukeRobotics/robosub-ros2/pull/37#pullrequestreview-2603920854

### PR DESCRIPTION
**Pull request by vedarshshah**
Closes #40.
- Added section about `SerialNode` class to README.
- Changed `--no-log-allocs` option in `test-thrusters` to `--log-allocs`, since repeatedly printing the same thruster allocs message is unnecessary in most cases.
- Cleaned up docstrings.
- Added error handling for edge cases when serial output is improperly formatted.
- In `peripheral.py`, handle cases where the Peripheral Arduino does not have any sensors or servos connected to it, when sensor or servo type is invalid, or when sensor or servo tag is a duplicate.
- Differentiated between discrete and continuous servos. Added `SetDiscreteServo` and `SetContinuousServo` service message types and removed `SetServo`. Updated `peripheral.py` to handle different servo types. Updated README with documentation about differentiation.
- In `serial_node.py`, made `max_consecutive_empty_lines` an argument passed to `init`.
- In `serial_node.py`, if an error is encountered when attempting to find the port of the serial device, then display a unique error message, different from when an error is encountered when connecting to that port.
- In `serial_node.py`, return a boolean to indicate if a write operation succeeded. In `peripheral.py`, send a "failed" response to service if `writeline` failed when sending servo command to Arduino.
- Added links from which we obtained Arduino libraries.
- Added comments in Arduino code.
- If invalid `ROBOT_NAME` is passed to Arduino, it will print an error message to serial instead of silently failing or assuming a default robot.
- Changed `PRESSURE_DELAY` and `SERVO_DELAY` to `50` so they don't run too frequently.
- Fixed critical bug in `DHT11.cpp` where it would get hung up on infinite loops in `readByte`. Added timeouts to loops in `readByte` to avoid hangups.
- Merged `MultiplexedServo` into `MultiplexedBasicESC` and combined it into just one file to make code more readable.

### CODE REVIEW

**Review by maxwellmlin — APPROVED** [review]
A few comments

File: onboard/src/offboard_comms/config/oogway.yaml [hunk 1]
```diff
@@ -22,12 +22,18 @@ arduino:
...
+          right: 1750
       - name: torpedoes
         tag: T
-        min_pwm: 700
-        max_pwm: 2300
+        service_name: /servos/torpedoes
+        type: discrete
+        states:
+          left: 1650
+          right: 1350
```
[the remark this answers is not in the document]
Comment (vedarshshah) [reply onboard/src/offboard_comms/config/oogway.yaml:36]: Made the states for the torpedo servo the same between `oogway` and `oogway-shell`.

Yes, the repetition isn't great, but I also don't have any good alternatives.

File: onboard/src/offboard_comms/sketches/peripheral/Oogway.cpp [hunk 2]
```diff
@@ -7,11 +7,8 @@
 
 class Oogway : public Robot {
 private:
+    // Flag to determine if this is Oogway's shell (true) or Oogway itself (false)
```
[the remark this answers is not in the document]
Comment (vedarshshah) [reply onboard/src/offboard_comms/sketches/peripheral/Oogway.cpp]: Currently, the flag doesn't change the behavior of the code, but is included to allow for any necessary changes in the future. Updated the comment to be more clear.

File: onboard/src/offboard_comms/offboard_comms/peripheral.py [hunk 3]
```diff
@@ -106,33 +171,72 @@ def process_line(self, line: str) -> None:
...
+
+            self.get_logger().error(error_msg)
+
+        return response
 
-                self.get_logger().error(error_msg)
+    def continuous_servo(self, request: SetContinuousServo.Request, response: SetContinuousServo.Response, tag: str) \
+            -> SetContinuousServo.Response:
+        """
+        Service callback to control discrete servos.
```
[the remark this answers is not in the document]
Comment (vedarshshah) [reply onboard/src/offboard_comms/offboard_comms/peripheral.py]: Updated
